### PR TITLE
feat: update first section of tutorial for v6

### PR DESCRIPTION
**Pull request by dannyhw**
Here's some initial proposed changes I'd like to make to the tutorial for v6.

The cli is still being updated for this to work but you can try it by checking out the cli locally from this branch

https://github.com/storybookjs/storybook/pull/20719

then run the cli like this inside the rn project you generate

/ABSOLUTE_PATH_TO_CODE/storybook/storybook/code/lib/cli/bin/index.js init --type react_native

### CODE REVIEW

File: content/intro-to-storybook/react-native/en/simple-component.md [hunk 1]
```diff
@@ -21,51 +21,101 @@ This process is similar to [Test-driven development](https://en.wikipedia.org/wi
...
 We’ll begin with a basic implementation of the `Task`, simply taking in the attributes we know we’ll need and the two actions you can take on a task (to move it between lists):
 
-```js:title=components/Task.js
-import * as React from 'react';
-import { TextInput, SafeAreaView } from 'react-native';
-import { styles } from '../constants/globalStyles';
+```jsx:title=components/Task.jsx
+import { StyleSheet, TextInput, View } from "react-native";
+
+const styles = StyleSheet.create({
```
Comment (dannyhw) [inline content/intro-to-storybook/react-native/en/simple-component.md:47]: pull out to global styles to simplify

File: content/intro-to-storybook/react-native/en/simple-component.md [hunk 2]
```diff
@@ -21,51 +21,101 @@ This process is similar to [Test-driven development](https://en.wikipedia.org/wi
...
+
+```jsx:title=components/Task.stories.jsx
+import { View } from "react-native";
+
+import { Task } from "./Task";
+
+export default {
+  title: "Task",
+  component: Task,
+  decorators: [
```
Comment (dannyhw) [inline content/intro-to-storybook/react-native/en/simple-component.md:82]: pull out decorator from initial version of story

File: content/intro-to-storybook/react-native/en/simple-component.md [hunk 3]
```diff
@@ -75,167 +125,108 @@ There are two basic levels of organization in Storybook: the component and its c
...
-When creating a story we use a base task (`task`) to build out the shape of the task the component expects. This is typically modelled from what the true data looks like. Again, `export`-ing this shape will enable us to reuse it in later stories, as we'll see.
+To define our stories we export an object with an `args` property. Arguments or [`args`](https://storybook.js.org/docs/react/writing-stories/args) for short, allow us to live-edit our components with the controls addon without restarting Storybook. Once an [`args`](https://storybook.js.org/docs/react/writing-stories/args) value changes, so does the component.
 
-<div class="aside">
-<a href="https://storybook.js.org/docs/react/essentials/actions"><b>Actions</b></a> help you verify interactions when building UI components in isolation. Oftentimes you won't have access to the functions and state you have in context of the app. Use <code>action()</code> to stub them in.
-</div>
+When creating a story, we use a base `task` arg to build out the shape of the task the component expects. Typically modeled from what the actual data looks like. Again, `export`-ing this shape will enable us to reuse it in later stories, as we'll see.
```
Comment (dannyhw) [inline content/intro-to-storybook/react-native/en/simple-component.md:140]: mention csf3

File: content/intro-to-storybook/react-native/en/composite-component.md [hunk 4]
```diff
@@ -18,303 +18,311 @@ Since `Task` data can be sent asynchronously, we **also** need a loading state t
...
+    // Shaping the stories through args composition.
+    // Inherited data coming from the Loading story.
+    ...Loading.args,
+    loading: false,
+  },
+};
 ```
 
-As you may have noticed, the `addDecorator()` was used in the previous chapter and in this one, it allows us to add some “context” to the rendering of each task. In this case we add padding around the list to make it easier to visually verify.
+As you may have noticed, a decorator was used in the previous chapter and in this one, it allows us to add some “context” to the rendering of each task. In this case we add padding around the list to make it easier to visually verify.
```
Comment (jonniebigodes) [inline content/intro-to-storybook/react-native/en/composite-component.md:135]: @dannyhw reframe this to include decorators per our conversation. We're not mentioning them in the "Simple Component" section", as we don't have a React Native version of the documentation; point at the React version of the documentation.

File: content/intro-to-storybook/react-native/en/simple-component.md [hunk 5]
```diff
@@ -75,167 +125,108 @@ There are two basic levels of organization in Storybook: the component and its c
...
 
-## Config
+Now that we've set up the basics lets re-run `yarn storybook` and see our changes.
```
Comment (jonniebigodes) [inline content/intro-to-storybook/react-native/en/simple-component.md:142]: @dannyhw, leave the config section in and reframe it, if possible to:
- Showcase how the React Native version detects the stories
- Refrain from mentioning deleting the default stories (leaving them in serves as guidelines for folks to see other examples).

File: content/intro-to-storybook/react-native/en/simple-component.md [hunk 6]
```diff
@@ -75,167 +125,108 @@ There are two basic levels of organization in Storybook: the component and its c
...
-An alternative way to achieve the same purpose is to use a JavaScript type system like TypeScript to create a type for the component properties.
-</div>
+![a gif showing the task component in storybook](/intro-to-storybook/react-native-task-component-completed.gif)
 
 ## Component built!
 
-We’ve now successfully built out a component without needing a server or running the entire frontend application. The next step is to build out the remaining Taskbox components one by one in a similar fashion.
+We’ve now successfully built out a component without needing a server or running the entire application. The next step is to build out the remaining Taskbox components one by one in a similar fashion.
 
 As you can see, getting started building components in isolation is easy and fast. We can expect to produce a higher-quality UI with fewer bugs and more polish because it’s possible to dig in and test every possible state.
```
Comment (jonniebigodes) [inline content/intro-to-storybook/react-native/en/simple-component.md:232]: End the chapter with an aside to tell the readers to commit their changes to their GitHub repo.

File: content/intro-to-storybook/react-native/en/composite-component.md [hunk 7]
```diff
@@ -18,303 +18,311 @@ Since `Task` data can be sent asynchronously, we **also** need a loading state t
...
+React Native storybook uses a script to generate imports for story files due to how dynamic imports are not supported in React Native. Lets run that script so our new story shows up:
 
-```javascript
-// storybook/config.js
-configure(() => {
-  require('../components/Task.stories.js');
-  require('../components/TaskList.stories.js');
-}, module);
+```sh
+yarn sb-rn-get-stories
```
Comment (jonniebigodes) [inline content/intro-to-storybook/react-native/en/composite-component.md:145]: @dannyhw tie in this snippet with some text / section similar to the "Simple Component" config section mentioning that with each new component/ story, the readers are required to run the command to allow Storybook to pull in the updated / new stories.

File: content/intro-to-storybook/react-native/en/composite-component.md [hunk 8]
```diff
@@ -18,303 +18,311 @@ Since `Task` data can be sent asynchronously, we **also** need a loading state t
...
-  });
-});
-```
-
-![TaskList test runner](/intro-to-storybook/tasklist-testrunner.png)
-
-Note that we’ve been able to reuse the `withPinnedTasks` list of tasks in both story and unit test; in this way we can continue to leverage an existing resource (the examples that represent interesting configurations of a component) in many ways.
-
-Notice as well that this test is quite brittle. It's possible that as the project matures, and the exact implementation of the `Task` changes --perhaps using a different styling prop or a `Text` rather than an `TextInput`--the test will fail, and need to be updated. This is not necessarily a problem, but rather an indication to be careful about liberally using unit tests for UI. They're not easy to maintain. Instead rely on visual, snapshot tests where possible.
+Note the position of the pinned item in the list. We want the pinned item to render at the top of the list to make it a priority for our users.
```
Comment (jonniebigodes) [inline content/intro-to-storybook/react-native/en/composite-component.md:328]: @dannyhw to avoid having the chapter end like this, as it seems rather abrupt, we can rephrase it to something like this:
"Success! We accomplished what we set out to do. If we check our updated UI, we can quickly see that our pinned tasks are now featured at the top of the list, matching the intended design. In the next chapters, we'll expand on what we've learned and continue adding complexity to our application by applying these principles to more complex UIs."

End with the GitHub aside mentioning to don't forget to commit the changes

**Review by jonniebigodes — CHANGES_REQUESTED** [review]
@dannyhw per our conversation, I left some items that require attention so we can publish the tutorial. Most of it is aimed what the get started section. As for the other two, you already have them mostly covered, and I left some minor tweaks to them. I'm going to brief our Design team to see when they can create the required assets to match this tutorial version.

When you can start addressing them, let me know, and I'll be more than glad to follow up with you and guide you on some other issues/questions you may have.

Looking forward to hearing from you and seeing the changes made.

Hope you have a great weekend.

Stay safe

File: content/intro-to-storybook/react-native/en/get-started.md [hunk 9]
```diff
@@ -4,503 +4,187 @@ tocTitle: 'Get started'
 description: 'Set up Storybook in your development environment'
 ---
 
-Storybook runs alongside your app in development mode. It helps you build UI components isolated from the business logic and context of your app. This edition of the Intro to Storybook tutorial is for React Native; other editions exist for [React](/intro-to-storybook/react/en/get-started), [Vue](/intro-to-storybook/vue/en/get-started), [Angular](/intro-to-storybook/angular/en/get-started), [Svelte](/intro-to-storybook/svelte/en/get-started) and [Ember](/intro-to-storybook/ember/en/get-started).
+Storybook helps you build UI components isolated from the business logic and context of your app. This edition of the Intro to Storybook tutorial is for React; other editions exist for [React](https://storybook.js.org/tutorials/intro-to-storybook/react/en/get-started/) [Vue](https://storybook.js.org/tutorials/intro-to-storybook/vue/en/get-started), [Angular](https://storybook.js.org/tutorials/intro-to-storybook/angular/en/get-started), [Svelte](https://storybook.js.org/tutorials/intro-to-storybook/svelte/en/get-started) and [Ember](https://storybook.js.org/tutorials/intro-to-storybook/ember/en/get-started).
```
[the remark this answers is not in the document]
Comment (dannyhw) [reply content/intro-to-storybook/react-native/en/get-started.md:7]: oops! yeah nice catch

File: content/intro-to-storybook/react-native/en/get-started.md [hunk 10]
```diff
@@ -4,503 +4,187 @@ tocTitle: 'Get started'
...
 
-<div class="aside">
-<p>We’ve used <code>svn</code> (Subversion) to easily download a folder of files from GitHub. If you don’t have subversion installed or want to just do it manually, you can grab the icons folder directly <a href="https://github.com/chromaui/learnstorybook-code/tree/master/src/assets">here</a> and the font <a href="https://github.com/google/fonts/tree/master/ofl/nunitosans">here</a>.</p></div>
+If what you see on your device matches this image then you've successfully setup storybook.
 
-```shell:clipboard=false
-svn export https://github.com/chromaui/learnstorybook-code/branches/master/src/assets/icon assets/icon
-svn export https://github.com/google/fonts/trunk/ofl/nunitosans assets/font
-```
+## Swap between storybook and app development
```
[the remark this answers is not in the document]
Comment (dannyhw) [reply content/intro-to-storybook/react-native/en/get-started.md:80]: I'm not sure cross-environment is the right wording here because that makes me think of ios/android or prod/staging. Is there another way we could word the heading? Maybe something like `How it works` or along those lines.

[the remark this answers is not in the document]
Comment (dannyhw) [reply content/intro-to-storybook/react-native/en/get-started.md:80]: Also I like the git section, however at this point we haven't made any changes so wouldn't there be nothing to commit?

[the remark this answers is not in the document]
Comment (dannyhw) [reply content/intro-to-storybook/react-native/en/get-started.md]: @jonniebigodes I've made most of these changes however I used `How it works`, let me know what you think

File: content/intro-to-storybook/react-native/en/get-started.md [hunk 11]
```diff
@@ -4,503 +4,187 @@ tocTitle: 'Get started'
...
-<div class="aside">
-  <p>During Storybook's install process, you'll be prompted to install react-native-server, do so as this package will help out immensely throughout the tutorial.</p>
-</div>
+Now that the project is created lets run the app to make sure everything is working as expected.
 
-### Set Up Jest with React Native
+You can pick ios or android, just run either and make sure the app is working.
 
-We have two out of three modalities configured in our app, but we still need one, we need to set up [Jest](https://facebook.github.io/jest/) to enable testing.
+For ios
```
[the remark this answers is not in the document]
Comment (dannyhw) [reply content/intro-to-storybook/react-native/en/get-started.md]: Done!

File: content/intro-to-storybook/react-native/en/composite-component.md (hunk 7 above)
Comment (dannyhw) [reply content/intro-to-storybook/react-native/en/composite-component.md]: I've done something similar to this

File: content/intro-to-storybook/react-native/en/composite-component.md (hunk 4 above)
Comment (dannyhw) [reply content/intro-to-storybook/react-native/en/composite-component.md]: Updated it

File: content/intro-to-storybook/react-native/en/composite-component.md (hunk 8 above)
Comment (dannyhw) [reply content/intro-to-storybook/react-native/en/composite-component.md]: updated

**Comment by dannyhw** [comment]
@jonniebigodes I've made the changes suggested, let me know what you think 🙏

File: content/intro-to-storybook/react-native/en/simple-component.md [hunk 12]
```diff
@@ -21,221 +21,168 @@ This process is similar to [Test-driven development](https://en.wikipedia.org/wi
...
+    "@storybook/addon-ondevice-actions",
+  ],
+};
+```
+
+If you check the `stories` property you'll see that Storybook is looking for stories in the `components` folder.
+
+In React Native Storybook we use the config in `main.js` to generate a file called `storybook.requires.js`, this is because React Native doesn't support dynamic imports yet. This file is used to load all the stories in the project and it also imports your addons.
+
+This file gets generated when you run `yarn storybook` to start storybook or `yarn storybook-generate` if you just want to regenerate the `storybook.requires.js` file. Whenever you find that a story is not being loaded, you can try regenerating this file and it should show up.
```
Comment (jonniebigodes) [inline content/intro-to-storybook/react-native/en/simple-component.md:40]: @dannyhw, can you combine this into a single paragraph that highlights the behavior and the caveat?

File: content/intro-to-storybook/react-native/en/simple-component.md [hunk 13]
```diff
@@ -21,221 +21,168 @@ This process is similar to [Test-driven development](https://en.wikipedia.org/wi
...
-To define our stories, we call `add()` once for each of our test states to generate a story. The action story is a function that returns a rendered element (i.e. a component class with a set of props) in a given state---exactly like a [Stateless Functional Component](https://reactjs.org/docs/components-and-props.html).
+- `component` -- the component itself
+- `title` -- how to refer to the component in the sidebar of the Storybook app
 
-When creating a story we use a base task (`task`) to build out the shape of the task the component expects. This is typically modelled from what the true data looks like. Again, `export`-ing this shape will enable us to reuse it in later stories, as we'll see.
-
-<div class="aside">
-<a href="https://storybook.js.org/docs/react/essentials/actions"><b>Actions</b></a> help you verify interactions when building UI components in isolation. Oftentimes you won't have access to the functions and state you have in context of the app. Use <code>action()</code> to stub them in.
-</div>
+Some optional values we also added were
```
Comment (jonniebigodes) [inline content/intro-to-storybook/react-native/en/simple-component.md:110]: @dannyhw drop the optional values and push the `argTypes` up as part of the construct of defining stories. The reasoning behind this is that some versions already use it and we can push the same way/narrative here to indirectly tell people that they should use it to define their stories. WDYT?

**Review by jonniebigodes — CHANGES_REQUESTED** [review]
@dannyhw this is shaping up nicely and hopefully, we'll get it out soon. Left a couple of items that require your attention. Let me know once you've addressed them. 

Hope you have a great day.

Stay safe

File: content/intro-to-storybook/react-native/en/get-started.md (hunk 10 above)
[the remark this answers is not in the document]
Comment (jonniebigodes) [reply content/intro-to-storybook/react-native/en/get-started.md]: Marking as solved. Agree with the reasoning and choice of words for this case. Thanks 🙏

**Review by jonniebigodes — APPROVED** [review]
